UnitTests/GitCommandsTests/Submodules/SubmoduleStatusProviderTests.cs

```diff
@@ -98,49 +98,31 @@ public void UpdateSubmoduleStatus_valid_result_for_second_nested_submodule()
             }
 
             [Test]
-            public void HasChangedToNone_valid_result()
+            public void Submodule_status_changes()
             {
-                SubmoduleTestHelpers.UpdateSubmoduleStatusAndWaitForResult(_provider, _repo3Module);
+                var currentModule = _repo1Module;
+                var result = SubmoduleTestHelpers.UpdateSubmoduleStructureAndWaitForResult(_provider, currentModule);
 
                 // No changes in repo
-                var changedFiles = GetStatusChangedFiles(_repo1Module);
+                var changedFiles = GetStatusChangedFiles(currentModule);
                 changedFiles.Should().HaveCount(0);
-                _provider.HasChangedToNone(changedFiles).Should().BeFalse();
+                SubmoduleTestHelpers.UpdateSubmoduleStatusAndWaitForResult(_provider, currentModule, changedFiles);
+                result.OurSubmodules.All(i => i.Detailed == null).Should().BeTrue();
 
                 // Make a change in repo2
                 _repo1.CreateFile(_repo2Module.WorkingDir, "test.txt", "test");
-                changedFiles = GetStatusChangedFiles(_repo1Module);
+                changedFiles = GetStatusChangedFiles(currentModule);
                 changedFiles.Should().HaveCount(1);
-                _provider.HasChangedToNone(changedFiles).Should().BeFalse();
+                SubmoduleTestHelpers.UpdateSubmoduleStatusAndWaitForResult(_provider, currentModule, changedFiles);
+                ////result.OurSubmodules[0].Detailed.IsDirty.Should().BeTrue();
```

Comment:
AsyncTestHelper.WaitForPendingOperations(); is not awaiting the status here, so the test fails (unless the debugger is connected).
I am guessing that is the test helper that is the problem.
Note: This was not tested before, not a new problem.

Reply:
Your message looks cryptic...
It sounds like you're having a problem. Could you please explain further?

Reply:
> It sounds like you're having a problem.

The test is not working. (The reason this is still WIP)

The testhelper is not awaiting the result.
See below for a commit (in  https://github.com/gerhardol/gitextensions/pull/new/feature/i6359-submodule-selective-update-1) where the test works, but definitely worse API.

Reply:
I had it enabled and it failed.
Then I force rebuilt and it worked...

Reply:
> I had it enabled and it failed.

Was this with this PR or the alternative branch?
I have not seen a solution for the testhelper or how to rewrite the tests

Reply:
Maybe @jbialobr or @sharwell can help...